src/models/StyledComponent.js

```diff
@@ -298,6 +323,17 @@ export default function createStyledComponent(target: Target, options: Object, r
   // $FlowFixMe
   WrappedStyledComponent.target = isTargetStyledComp ? target.target : target;
 
+  // $FlowFixMe
+  const targetDefaultProps = target.targetDefaultProps || {};
+  // $FlowFixMe
+  const defaultProps = target.defaultProps || {};
+  // $FlowFixMe
+  WrappedStyledComponent.targetDefaultProps = {
+    ...targetDefaultProps,
+    ...defaultProps,
+    ...mergeStyleProp(defaultProps.style, targetDefaultProps.style),
```

Comment:
I don't think there's a need for this separate util, let's just make this

```suggestion
...({ ...defaultProps.style, ...targetDefaultProps.style })
```

Reply:
This wouldn't work. It's missing the `style` key, we can't spread `undefined` and we can't send `style: undefined` or `style: {}`. I change the util for mergeProp, so we can use it for other special case. I could use reduce, or lodash.get equivalent but I think this way is faster.

Reply:
Ah I see, okay let's just keep i this way then! Good catch